nat-lab/tests/uniffi/serialization.py

```diff
@@ -0,0 +1,62 @@
+import inspect
+import Pyro5.api  # type: ignore
+
+
+def init_serialization(libtelio):
+    all_members = inspect.getmembers(libtelio, inspect.isclass)
+    public_members = []
+    for name, cls in all_members:
+        if not name.startswith("_"):
+            public_members.append(cls)
+            nested_members = inspect.getmembers(cls, inspect.isclass)
+            for name, cls in nested_members:
+                if not name.startswith("_"):
+                    public_members.append(cls)
+
+    classes = tuple(
+        filter(
+            lambda m: str(m).startswith("<class ") and libtelio.__name__ in str(m),
+            public_members,
+        )
+    )
+    enums = tuple(filter(lambda m: str(m).startswith("<enum "), public_members))
+
+    def serialize_obj(obj):
+        if isinstance(obj, classes):
+            class_name = type(obj).__name__
+            out = {
+                "__class__": class_name,
+                "__dict__": {k: serialize_obj(v) for k, v in obj.__dict__.items()},
+            }
+            return out
+        if isinstance(obj, enums):
+            enum_name = type(obj).__name__
+            return {"__enum__": enum_name, "value": obj.name}
+        if isinstance(obj, list):
```

Comment:
What if an object is neither ?

Reply:
Then it it passed over the Pyro5 boundary as-is. The intention for it is to only apply to primitive types. The only type which may become relevant to add here at some point is `dict`, but we don't have any such cases in the udl file right now so I skipped adding it here